shoop/campaigns/models/context_conditions.py

```diff
@@ -27,20 +27,19 @@ class ContactGroupCondition(ContextCondition):
     model = ContactGroup
     identifier = "contact_group_condition"
     name = _("Contact Group")
-    description = _("Contact group")
 
     contact_groups = models.ManyToManyField(ContactGroup, verbose_name=_("contact groups"))
 
+    def matches(self, context):
+        if not context.customer:
+            return False
+        customers_groups = context.customer.groups.all()
+        return self.contact_groups.filter(pk__in=customers_groups).exists()
```

Comment:
This does not work for the AnonymousContact, since the are falsy objects and the code will go to `if not context.customer` branch rather than checking the anonymous contacts group as it should. The `if not context.customer` should probably be just removed or changed to:
```python
customer = (context.customer or AnonymousContact())
```
and then use `customer.groups.all()` below.